scripts/shared/e2e.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+## Process command line flags ##
+
+source /usr/share/shflags/shflags
+DEFINE_string 'cluster_settings' '' "Settings file to customize cluster deployments"
+DEFINE_string 'focus' '.*' 'Ginkgo focus for the E2E tests'
+FLAGS "$@" || exit $?
+eval set -- "${FLAGS_ARGV}"
+
+focus="${FLAGS_focus}"
+cluster_settings="${FLAGS_cluster_settings}"
+
+set -em -o pipefail
+
+source ${SCRIPTS_DIR}/lib/debug_functions
+source ${SCRIPTS_DIR}/lib/utils
+
+# Always source the shared cluster settings, to set defaults in case something wasn't set in the provided settings
+source "${SCRIPTS_DIR}/lib/cluster_settings"
+[[ -z "${cluster_settings}" ]] || source ${cluster_settings}
+
+### Functions ###
+
+function deploy_env_once() {
+    if with_context cluster3 kubectl wait --for=condition=Ready pods -l app=submariner-engine -n "${SUBM_NS}" --timeout=3s > /dev/null 2>&1; then
+        echo "Submariner already deployed, skipping deployment..."
+        return
+    fi
```

Comment:
Why is this necessary since the `e2e` make target depends/should depend on the `deploy` target?  Also this assumes that submariner is actually deployed by the project invoking it (admiral does not).

Reply:
Yeah it will mostly be true.. It was part of the "lazy deploy" change on submariner https://github.com/submariner-io/submariner/pull/512

However I can add a flag to toggle this, if that makes sense let me know and I'll add one

Reply:
But why have this here at all? How does it differ from or is better than a make target dependency?

Reply:
This was already discussed on https://github.com/submariner-io/submariner/pull/488 (which you approved) and superseded by https://github.com/submariner-io/submariner/pull/510 (which was itself superseded by https://github.com/submariner-io/submariner/pull/512).

The gist is: if it's a `make` dependency, then running `make e2e` would cause a whole chain of actions (compile, docker image, clusters) to occur even if you only intend to run only the e2e tests since you only changed the files there.
Hence this seemed like a nice way to avoid a hard dependency of `e2e` on `deploy` and allow the requested behavior of running just the e2e tests.

I can leave this code solely in submariner, but I think most projects could benefit from it.

If we have `e2e` depend on `delpoy` in the Makefile, it'll be very hard not to rerun `deploy` each time you run `make e2e` since the `deploy` target itself isnt a real file so make can't know if it should or shouldn't re run it.

Hence I proposed this can be toggle-able via a flag to this script.

Reply:
Yeah, we need to optimize that make e2e so the workflow of developing a new e2e test is not a PITA for the developer, or involves logging into the dapper shell to run the e2e tests manually or other kind of setup.

Also having a common "e2e" that will let you re-run your e2e tests in a simple way is better because we can have a single development document then.

I see it's not ideal. May be there's a more appropriate name than "lazy" ? :)